tests/Fixtures/inc/Engine/License/API/PricingClient/getPricingData.php

```diff
@@ -5,47 +5,110 @@
 $data = json_decode( $json );
 
 return [
-	'testShouldReturnFalseWhenWPError' => [
+	'testShouldReturnDataWhenCached' => [
 		'config'   => [
-			'transient' => false,
-			'response'  => new WP_Error( 'http_request_failed', 'error' ),
+			'pricing-transient' => true,
+			'timeout-active'    => false,
+			'timeout-duration'  => false,
+			'response'          => false,
 		],
-		'expected' => false,
+		'expected' => [
+			'result' => $data,
+		]
 	],
-	'testShouldReturnFalseWhenNot200'  => [
+
+	'testShouldReturnFalseWhenTimeoutActive' => [
+		'config'   => [
+			'pricing-transient' => false,
+			'timeout-active'    => true,
+			'timeout-duration'  => false,
+			'response'          => false,
+		],
+		'expected' => [
+			'result' => false,
+		],
+	],
+
+	'testReturnFalseWhenWPError' => [
+		'config'   => [
+			'pricing-transient' => false,
+			'timeout-active'    => false,
+			'timeout-duration'  => false,
+			'response'          => new WP_Error( 'http_request_failed', 'error' ),
+		],
+		'expected' => [
+			'result' => false,
+		],
+	],
+
+	'testShouldReturnFalseWhenNot200' => [
 		'config'   => [
-			'transient' => false,
-			'response'  => [
+			'pricing-transient' => false,
+			'timeout-active'    => false,
+			'timeout-duration'  => false,
+			'response'          => [
 				'code' => 404,
 				'body' => false,
 			],
 		],
-		'expected' => false,
+		'expected' => [
+			'result' => false,
+		],
 	],
-	'testShouldReturnFalseWhenNoBody'  => [
+
+	'testShouldReturnFalseWhenNoBody' => [
 		'config'   => [
-			'transient' => false,
-			'response'  => [
+			'pricing-transient' => false,
+			'timeout-active'    => false,
+			'timeout-duration'  => false,
+			'response'          => [
 				'code' => 200,
 			],
 		],
-		'expected' => false,
-	],
-	'testShouldReturnDataWhenCached'   => [
-		'config'   => [
-			'transient' => true,
-			'response'  => false,
+		'expected' => [
+			'result' => false,
 		],
-		'expected' => $data,
 	],
-	'testShouldReturnDataWhenSuccess'  => [
+
+	'testShouldReturnDataWhenSuccess' => [
 		'config'   => [
-			'transient' => false,
-			'response'  => [
+			'pricing-transient' => false,
+			'timeout-active'    => false,
+			'timeout-duration'  => false,
+			'response'          => [
 				'code' => 200,
 				'body' => $json,
 			],
 		],
-		'expected' => $data,
+		'expected' => [
+			'result' => $data,
+		],
+	],
+
+	'testShouldDoubleTimeoutDurationFromPreviousDuration' => [
+		'config'   => [
+			'pricing-transient' => false,
+			'timeout-active'    => false,
+			'timeout-duration'  => 300,
+			'response'          => [ 'code' => 404 ],
+		],
+		'expected' => [
+			'result'           => false,
+			'timeout-duration' => 600
+		],
+	],
+
+	'testShouldNotSetTimeoutDurationLongerThanADay' => [
+		'config'   => [
+			'pricing-transient' => false,
+			'timeout-active'    => false,
+			'timeout-duration'  => rocket_get_constant( 'DAY_IN_SECONDS' )
+								   - rocket_get_constant( 'HOuR_IN_SECONDS' ),
```

Comment:
@iCaspar For consistency, shouldn't 'HOuR_IN_SECONDS' be 'HOUR_IN_SECONDS'

Reply:
Yes, it should be all caps here. 😳 
Also, yes, it is intended that once the timeout reaches 1 day it remains at 1 day until the API call is successful and the transient is deleted.